ReleaseProcessAutomation/Steps/PipelineSteps/ReleasePatchStep.cs

```diff
@@ -52,36 +53,39 @@ public class ReleasePatchStep : ReleaseProcessStepBase, IReleasePatchStep
       IInputReader inputReader,
       IMSBuildCallAndCommit msBuildCallAndCommit,
       IContinueReleasePatchStep continueReleasePatchStep,
+      IPushNewReleaseBranchStep pushNewReleaseBranchStep,
       IAnsiConsole console,
       IJiraFunctionality ijIraFunctionality)
       : base(gitClient, config, inputReader, console)
   {
     _msBuildCallAndCommit = msBuildCallAndCommit;
     _continueReleasePatchStep = continueReleasePatchStep;
     _ijIraFunctionality = ijIraFunctionality;
+    _pushNewReleaseBranchStep = pushNewReleaseBranchStep;
   }
 
   public void Execute (SemanticVersion nextVersion, string? commitHash, bool startReleasePhase, bool pauseForCommit, bool noPush, bool onMaster)
   {
     EnsureWorkingDirectoryClean();
 
-    if (GitClient.GetCurrentBranchName() == null)
+    var currentBranchName = GitClient.GetCurrentBranchName();
+    if (currentBranchName == null)
     {
       const string message = "Could not get the name of the current branch.";
       throw new InvalidOperationException(message);
     }
 
     if (onMaster)
     {
-      if (!GitClient.IsOnBranch("master"))
+      if (!currentBranchName.Equals("master"))
       {
         const string message = "Cannot release a patch for master when not on master branch.";
         throw new InvalidOperationException(message);
       }
     }
     else
     {
-      if (!GitClient.IsOnBranch("hotfix/"))
+      if (!currentBranchName.StartsWith("hotfix/"))
```

Comment:
This should also be changed to `==`

Reply:
Im not sure youre correct here, we dont know which version number will be appended to the hotfix/*** so we cant really check for equality

Reply:
I should learn to read :D